src/main/scala/com/lightbend/kafka/scala/streams/KStreamS.scala

```diff
@@ -57,12 +58,14 @@ class KStreamS[K, V](val inner: KStream[K, V]) {
     inner.branch(predicates.map(_.asPredicate): _*).map(kstream => wrapKStream(kstream))
   }
 
-  def through(topic: String): KStreamS[K, V] = inner.through(topic)
+  def through(topic: String)(implicit keySerde: Serde[K], valueSerde: Serde[V]): KStreamS[K, V] =
```

Comment:
The idea behind this version of the API is to allow the user to use the default serializers, deserializers and the producer's DefaultPartitioner. Here's the Javadoc ..

> Materialize this stream to a topic and creates a new KStream from the topic using default serializers and deserializers and producer's DefaultPartitioner. 

By asking the user to supply implicit parameters for them makes the contract contradict the original API intent. And also the API variant with an explicit `Produced` parameter is also present here.

If you look at the API design scheme of Kafka Streams 1.0, clearly they have 2 variants - **the first** that takes these arguments from the default setups and the **second one** that allows the user to *explicitly* pass them through `Produced`, `Serialized`, `Consumed` abstractions. In the version that u suggest, even though the user may have the defaults set for the serializers, she will have to pass them again as implicits.

Reply:
I would argue that typeclass implicits are both a better solution and more common in scala land (see e.g. play-json `Format` or play `ConfigLoader` and `ContentWritable`). The configuration file solution should be available, but be secondary and not-recommended in my opinion as they carry the risk of runtime errors. Maybe we can expose them under alternative method names?

> By asking the user to supply implicit parameters for them makes the contract contradict the original API intent.
You could argue that the default `Serde`'s from the config are a java mechanism to prevent having to explicitly pass them all over the place, for lack of implicit parameters in java. Replacing this mechanism with scala implicits, I'd say that the proposed change nicely 'translates' this intent to Scala. With the added benefit that we avoid runtime errors when the default serializer is not able to handle the type.

Reply:
> typeclass implicits are both a better solution and more common in scala land

fully agree but here by introducing implicits we are losing a big use case - defining default (de-)serializers upfront and then using the APIs like `through` or `to` with just the topic specified as argument. And this is a very well established idiom when programming with kafka-streams. These users will have to define implicits for them and ensure they are in scope. And this they need to do in every function that use these APIs.

Reply:
Implicit conversion  should be provided by this package.
Then user instead of setting default we'll be doing import sth.implicits._
It's not a big deal.
Please look how it's done in Apache Spark.

Reply:
https://spark.apache.org/docs/2.2.1/api/java/org/apache/spark/sql/SparkSession.implicits$.html